tests/SpecTests/ClientSideEncryption/Prose22_RangeExplicitEncryptionTest.php

```diff
@@ -0,0 +1,479 @@
+<?php
+
+namespace MongoDB\Tests\SpecTests\ClientSideEncryption;
+
+use ArrayIterator;
+use Generator;
+use Iterator;
+use MongoDB\BSON\Binary;
+use MongoDB\BSON\Decimal128;
+use MongoDB\BSON\Document;
+use MongoDB\BSON\UTCDateTime;
+use MongoDB\Driver\ClientEncryption;
+use MongoDB\Driver\Exception\EncryptionException;
+use MultipleIterator;
+
+use function base64_decode;
+use function file_get_contents;
+use function get_debug_type;
+use function is_int;
+use function version_compare;
+
+/**
+ * Prose test 22: Range Explicit Encryption
+ *
+ * @see https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests/README.rst#range-explicit-encryption
+ * @group csfle
+ * @group serverless
+ */
+class Prose22_RangeExplicitEncryptionTest extends FunctionalTestCase
+{
+    private $clientEncryption;
+    private $collection;
+    private $encryptedClient;
+    private $key1Id;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        if ($this->isStandalone() || ($this->isShardedCluster() && ! $this->isShardedClusterUsingReplicasets())) {
+            $this->markTestSkipped('Range explicit encryption tests require replica sets');
+        }
+
+        if (version_compare($this->getServerVersion(), '7.0.0', '<')) {
+            $this->markTestSkipped('Range explicit encryption tests require MongoDB 7.0 or later');
+        }
+
+        $client = static::createTestClient();
+
+        $key1Document = $this->decodeJson(file_get_contents(__DIR__ . '/../client-side-encryption/etc/data/keys/key1-document.json'));
+        $this->key1Id = $key1Document->_id;
+
+        // Drop the key vault collection and insert key1Document with a majority write concern
+        self::insertKeyVaultData($client, [$key1Document]);
+
+        $this->clientEncryption = $client->createClientEncryption([
+            'keyVaultNamespace' => 'keyvault.datakeys',
+            'kmsProviders' => ['local' => ['key' => new Binary(base64_decode(self::LOCAL_MASTERKEY), 0)]],
+        ]);
+
+        $autoEncryptionOpts = [
+            'keyVaultNamespace' => 'keyvault.datakeys',
+            'kmsProviders' => ['local' => ['key' => new Binary(base64_decode(self::LOCAL_MASTERKEY), 0)]],
+            'bypassQueryAnalysis' => true,
+        ];
+
+        $this->encryptedClient = self::createTestClient(null, [], [
+            'autoEncryption' => $autoEncryptionOpts,
+            /* libmongocrypt caches results from listCollections. Use a new
+             * client in each test to ensure its encryptedFields is applied. */
+            'disableClientPersistence' => true,
+        ]);
+    }
+
+    public function setUpWithTypeAndRangeOpts(string $type, array $rangeOpts): void
+    {
+        if ($type === 'DecimalNoPrecision' || $type === 'DecimalPrecision') {
```

Comment:
With private consts, you can be sure that there is no typo in this strings.

Reply:
Will do. This also made me realize that we can replace most of the existing `private static` class properties with private constants. Opened [PHPLIB-1152](https://jira.mongodb.org/browse/PHPLIB-1152) to track that.